ast/Trees.cc

```diff
@@ -1411,6 +1414,31 @@ string BlockArg::nodeName() {
 ParsedFilesOrCancelled::ParsedFilesOrCancelled() : trees(nullopt){};
 ParsedFilesOrCancelled::ParsedFilesOrCancelled(std::vector<ParsedFile> &&trees) : trees(move(trees)) {}
 
+ParsedFilesOrCancelled ParsedFilesOrCancelled::cancel(std::vector<ParsedFile> &&trees, WorkerPool &workers) {
+    if (!trees.empty()) {
+        absl::BlockingCounter threadBarrier(std::max(workers.size(), 1));
```

Comment:
Note: I'm pretty sure `workers.size()` returns 0 when threading is disabled, hence the `std::max` with `1`. I can add a comment if desired.

Reply:
Comment wouldn't hurt, I spose!